kernel/tests/dv.rs

```diff
@@ -47,3 +84,387 @@ fn non_dv_table() -> Result<(), Box<dyn std::error::Error>> {
     assert_eq!(total_rows, 10);
     Ok(())
 }
+
+/// Helper to extract scan files from a snapshot
+fn get_scan_files(
+    snapshot: Arc<Snapshot>,
+    engine: &dyn delta_kernel::Engine,
+) -> DeltaResult<Vec<FilteredEngineData>> {
+    let scan = snapshot.scan_builder().build()?;
+    let all_scan_metadata: Vec<_> = scan.scan_metadata(engine)?.collect::<Result<Vec<_>, _>>()?;
+
+    Ok(all_scan_metadata
+        .into_iter()
+        .map(|sm| sm.scan_files)
+        .collect())
+}
+
+/// Helper to create add file metadata for multiple files.
+/// Each tuple contains: (file_path, file_size, num_records)
+fn create_add_files_metadata(
```

Comment:
this seems like it already exists in write.rs, please find a way to remove the duplication.